src/Bridge/Doctrine/DBAL/Types/AbstractEnumType.php

```diff
@@ -14,12 +14,17 @@
 use Doctrine\DBAL\Types\Type;
 use Elao\Enum\EnumInterface;
 
+/**
+ * @template T
```

Comment:
I think the `T` should be an `EnumInterface` so for `Gender` enum it would look like:

```php
/**
 * @extends AbstractEnumType<Gender>
 */
class GenderType extends AbstractEnumType
```

then the `getEnumClass()` should `@psalm-return class-string<T>`

Reply:
Indeed! Thanks!